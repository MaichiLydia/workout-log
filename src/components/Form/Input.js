import React from 'react'
import { isGreaterDate, timerValidate } from '../../helpers'
import PropTypes from 'prop-types'

function Input(props){

    function inputValidate(e){ 
        const name = e.target.name;
        const value = e.target.value;


        if(props.required && value.trim() === ''){
            props.onChange(name, value, "Required Field")
            return
        }

        if(props.type === 'time' && !(timerValidate(e.target.value, props.min))){
            props.onChange(name, value, e.target.name + ' must be greater than ' + props.min )
            return
        }

        if(props.type === 'date' && !(isGreaterDate(e.target.value, props.min)) ){
            props.onChange(name, value, " Date must be greater than 01 Jan 2000");
            return
        }
        props.onChange(name,value);
    }

    if(props.type === 'date' || props.type === 'time'){
        return (
        <input 
            className="form__input"
            id={props.id}
            type={props.type}
            name={props.name}
            placeholder={props.placeholder}
            min={props.min}
            max={props.max}
            onBlur={inputValidate}
             />

        )       
    }
    return (
       
        <input 
            className="form__input"
            id={props.id}
            type={props.type}
            name={props.name}
            min={props.min}
            max={props.max}
            placeholder={props.placeholder}
            onChange={inputValidate} />
    )
}

Input.propTypes = {
    id: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    required: PropTypes.bool,
    min: PropTypes.string,
    max: PropTypes.string,
    placeholder: PropTypes.string,
    onChange: PropTypes.func

}
export default Input;
