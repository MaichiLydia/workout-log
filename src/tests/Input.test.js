import React from 'react';
import Input from '../components/Form/Input';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('check if input receives the right values', () => {
    const props = {
        id: "name",
        type: "text",
        name: "name",
        required: true,
        placeholder: "Digite o texto aqui",
        onChange: () => true
        }
    const input = mount(<Input {...props}/>)
    expect(toJson(input)).toMatchSnapshot()
});


