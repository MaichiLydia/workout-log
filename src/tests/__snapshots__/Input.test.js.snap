// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if input receives the right values 1`] = `
<Input
  id="name"
  name="name"
  onChange={[Function]}
  placeholder="Digite o texto aqui"
  required={true}
  type="text"
>
  <input
    className="form__input"
    id="name"
    name="name"
    onChange={[Function]}
    placeholder="Digite o texto aqui"
    type="text"
  />
</Input>
`;
